src/wallet/wallet.cpp

```diff
@@ -1740,6 +1740,18 @@ bool CWalletTx::InMempool() const
     return false;
 }
 
+uint64_t CWalletTx::GetMempoolAncestorCount() const
+{
+    LOCK(mempool.cs);
+    return mempool.mapTx.find(GetHash())->GetCountWithAncestors();
+}
+
+uint64_t CWalletTx::GetMempoolDescendantCount() const
+{
+    LOCK(mempool.cs);
+    return mempool.mapTx.find(GetHash())->GetCountWithDescendants();
+}
```

Comment:
Agree with @luke-jr, but I think we need something higher-level anyway (see further).